### ISSUE TITLE
search: showing "more" facet values

### ISSUE THREAD

**Issue by tiborsimko** [cernopendata/opendata.cern.ch#1833]
(1) Note that we have many records with `distribution.formats` equal to `root`.  Yet, when someone launches an empty search, they don't appear in the file type facet list offered by default:

![file-types](https://user-images.githubusercontent.com/517546/33020454-aaf5db46-cdfe-11e7-845a-9fd35a2f592a.png)

The "root" value is not displayed there, even though it is relatively popular.  Moreover, the users don't have any way to reach it.

(2) UX-wise, it is even more surprising when someone looks for collision datasets directly:

- http://0.0.0.0/search?page=1&size=20&q=&type=Dataset&subtype=Collision

where the user currently sees:

![file-types-collision-datasets](https://user-images.githubusercontent.com/517546/33020613-5196c974-cdff-11e7-9fe5-1ff3860f8910.png)

yet without a possibility to see "root" format that is native for the collision datasets.

(3) We should make sure that the non-empty facet values can be reached in some way:

- e.g. consider displaying non-empty facet values first; (but we discussed alphabetical order rather than the popularity order before?!)  
- e.g. consider offering buttons to list "more" and "less" of facet values, as we used to do on [COD2 search pages](http://opendata.cern.ch/search?p=cms).

CC @annatrz @daslerr @sefeg

**Comment by daslerr** [comment]
Part of why we discussed alphabetical order was so the facets wouldn't change places in the list when the result values were recalculated. I still think this is a good idea. So my vote is for having a "more" button.

**Comment by tiborsimko** [comment]
@daslerr Yes, a "more" button would be nice to have in any case, so that people could "browse" in the value list, as it were. But I wonder whether the second situation above is really UX-friendly with the current showing-zeros approach: the  users may be thinking that the "more" button will give them "more of zeros", which they are not really interested in seeing 😄 In my eyes, for some facets, where the value list is limited (such as "experiment"), it is not that much of a problem to display empty values; while for other facets, where the value list is long (such as "file types"), it may be not very user-friendly even with the more/less buttons, precisely because we shall have lots of zeros "in the middle" of interesting values.  E.g. imagine theh "keyword" facet where we may have 300+ values soon...

**Comment by daslerr** [comment]
Between the two options presented above, I prefer number 2. But I do think that returning to the practice of not showing facets that have 0 results could be less confusing. We can discuss with @sefeg tomorrow to make sure we're considering all implications and then decide.

**Comment by ioannistsanaktsidis** [comment]
Is this one considered done ?

**Comment by sefeg** [comment]
I agree with @daslerr that we should go for option 2! I also think that we should avoid creating facets that have a huge amount of elements assigned ("300+").

This can then become part of the next user testing iteration to verify that the selected option is usable.

Thinking about the future (after our release) we might also want to muse about spending time on correctly calculating the values of ALL (not only the AND-related) facet elements. At that point we could also muse about hiding elements that would definitely not return any additional results.

**Comment by tiborsimko** [comment]
> Is this one considered done ?

Nope, since there is still no way to get to "root" facet value.  

It may be especially confusing to people that we display different number of values in different facets, and not a fixed maximum value of say 5 facets.  Hence, seeing say "energy" facet with 3 values, "topic category" facet with 6 values, and "file type" facet with 10 values, people may be led to think that there are no more file types values (such as "root"), as in the first two cases.

Having more/less facet value buttons would solve this unambiguously.

> I also think that we should avoid creating facets that have a huge amount of elements assigned ("300+").

They are be especially useful to navigate in the results sea. Think the "authors" facet on INSPIRE for example, when a user would like to explore a topic such as neutrino oscillations, and see which authors publish on this topic the most.  On COD, we don't have much individual authors, but we have keywords, file types, etc that can grow in numbers.

**Comment by tiborsimko** [comment]
@sefeg @pamfilos @ioannistsanaktsidis As an alternative to more/less facet value list buttons, we could also do something ultra simple on the UX/UI side, and present facet value lists in a rather more compressed manner, so that intermediate empty lines wouldn't eat up so much vertical space. This might permit us to display easily more values than ~10 for the given facet, alleviating the need for more/less button functionality. 

See for example:

- https://libraries.io/search?q=flask

where quite a lot of facet values are displayed, yet without a need for more/less facet buttons.

TL;DR Showing ~15 instead of ~10 values in a more compact manner could hopefully make all the "root" and "raw" facet value visible 😄

**Comment by sefeg** [comment]
Yes, @tiborsimko we can certainly show more options most easily by reducing the white space between the facet elements (as has been done on the page you linked earlier).

**Comment by annatrz** [comment]
So for now we'll show all facets values, and just change the way how they're displayed (reducing space between them)
